lib/Elastica/Index.php

```diff
@@ -295,7 +293,7 @@ public function exists()
      */
     public function createSearch($query = '', $options = null)
     {
-        $search = new Search($this->getClient());
+        $search = new Search($this->getClient(), $this->getClient()->getResultSetBuilder());
```

Comment:
I've removed knowledge of the Builder from the Client. Now, the only way to configure which Builder is used is by passing it into a Search constructor or via the proxy createSearch() method on an Index or a Type.